frontend/apps/web/app/(mgmt)/[account]/jobs/[id]/source/components/DataSyncConnectionCard.tsx

```diff
@@ -625,21 +625,10 @@ export default function DataSyncConnectionCard({ jobId }: Props): ReactElement {
                 form.setValue('destinationOptions', updatedDestOpts);
               }}
               onEditMappings={(values) => {
-                const valuesMap = new Map(
-                  values.map((v) => [
-                    `${v.schema}.${v.table}.${v.column}`,
-                    v.transformer,
-                  ])
-                );
+                const valuesMap = new Map(values.map((v) => [`${v.id}`, v]));
                 formMappings.forEach((fm, idx) => {
-                  const fmKey = `${fm.schema}.${fm.table}.${fm.column}`;
-                  const fmTrans = valuesMap.get(fmKey);
-                  if (fmTrans) {
-                    update(idx, {
-                      ...fm,
-                      transformer: fmTrans,
-                    });
-                  }
+                  const value = valuesMap.get(fm.id);
+                  update(idx, value ?? fm);
```

Comment:
Generally we want to conditionally call update if the fm has the value being updated so we can avoid calling `update` unless we absolutely need to.

Reply:
updated